test/e2e/snaps/ppom-blockaid-alert.spec.js

```diff
@@ -0,0 +1,192 @@
+            params,
+          });
+          await driver.executeScript(
+            `window.transactionHash = window.ethereum.request(${request})`,
+          );
+
+          // Wait for confirmation pop-up
+          const windowHandles = await getWindowHandles(driver, 3);
+          await driver.switchToWindow(windowHandles.popup);
+
+          const isPresent = await driver.isElementPresent(bannerAlertSelector);
+          assert.equal(
+            isPresent,
+            false,
+            `Banner alert unexpectedly found. \nExpected detail: ${logExpectedDetail}`,
+          );
+
+          // Wait for confirmation pop-up to close
+          await driver.clickElement({ text: 'Reject', tag: 'button' });
+          await driver.switchToWindow(windowHandles.dapp);
+        }
+      },
+    );
+  });
+
+  /**
+   * Disclaimer: this test may be missing checks for some reason types. e.g. blur, domain, and failed
+   */
+  it('should show security alerts for malicious requests', async function () {
+    await withFixtures(
+      {
+        dapp: true,
+        fixtures: new FixtureBuilder()
+          .withNetworkController(mainnetProviderConfig)
+          .withPermissionControllerConnectedToTestDapp()
+          .withPreferencesController({
+            securityAlertsEnabled: true,
+          })
+          .build(),
+        defaultGanacheOptions,
+        title: this.test.title,
+      },
+
+      async ({ driver }) => {
+        await driver.navigate();
+        await unlockWallet(driver);
+        await openDapp(driver);
+
+        const expectedTitle = 'This is a deceptive request';
+
+        const testMaliciousConfigs = [
+          {
+            btnSelector: '#maliciousPermit',
+            expectedDescription:
+              'If you approve this request, a third party known for scams might take all your assets.',
+            expectedReason: 'permit_farming',
+          },
+          {
+            btnSelector: '#maliciousSeaport',
+            expectedDescription:
+              'If you approve this request, someone can steal your assets listed on OpenSea.',
+            expectedReason: 'seaport_farming',
+          },
+          {
+            btnSelector: '#maliciousTradeOrder',
+            expectedDescription:
+              'If you approve this request, you might lose your assets.',
+            expectedReason: 'trade_order_farming',
+          },
+        ];
+
+        for (const config of testMaliciousConfigs) {
+          const { expectedDescription, expectedReason, btnSelector } = config;
+
+          // Click TestDapp button to send JSON-RPC request
+          await driver.clickElement(btnSelector);
+
+          // Wait for confirmation pop-up
+          const windowHandles = await getWindowHandles(driver, 3);
+          await driver.switchToWindow(windowHandles.popup);
+
+          const bannerAlert = await driver.findElement(bannerAlertSelector);
+          const bannerAlertText = await bannerAlert.getText();
```

Comment:
You're introducing a race condition here, see this Slack message from Mark Stacey: https://consensys.slack.com/archives/GTQAGKY5V/p1686238528522549

Reply:
nice catch! thank you @HowardBraham

Reply:
fixed [d5b5d43](https://github.com/MetaMask/metamask-extension/pull/21029/commits/d5b5d43bde4f0ab84290f1c9ea8dbf06aedf031e)